### PR TITLE
Edit exception message when editing certain fields of a catalogue item with children

### LINKED ISSUE #353: Message when attempting to edit catalogue item fields with children is misleading
`Catalogue item has child elements and cannot be updated`  is returned when changing the manufacturer id or properties - ideally this would actually say that its just these fields that can't be updated instead.

### PR DESCRIPTION
**Pull request by rowan04**
## Description

Edit the returned exception message to make it clearer when trying to update protected fields on a catalogue item or catalogue category that has a child element.

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

Resolves #353

### CODE REVIEW

**Review by joelvdavies — COMMENTED** [review]
General comment could you perhaps list the fields from `CATALOGUE_ITEM_WITH_CHILD_NON_EDITABLE_FIELDS` so it's kept up to date with changes. Also I just noticed we also have exactly the same sort of error message for the catalogue category edit could you update these as well so they are consistent.

The front end may need a change to be more user friendly
image

**Comment by rowan04** [comment]
The linter check is failing, but I'm pretty sure all the lines I added are under 120. So is this failing because of previously written lines in the files I edited?

File: test/e2e/test_catalogue_category.py [hunk 1]
```diff
@@ -987,7 +987,9 @@ def test_partial_update_non_leaf_to_leaf_when_has_child_catalogue_category(self)
         self.patch_catalogue_category(catalogue_category_id, {"is_leaf": True})
 
         self.check_patch_catalogue_category_failed_with_detail(
-            409, "Catalogue category has child elements and cannot be updated"
+            409,
+            "Catalogue category has child elements, so the following fields cannot be updated: "
+                + "is_leaf, properties"
```
[the remark this answers is not in the document]
Comment (joelvdavies) [reply test/e2e/test_catalogue_category.py:992]: Yeah I would use the variable its simple enough and it ensures future additions will require minimal changes.

**Comment by joelvdavies** [comment]
> The linter check is failing, but I'm pretty sure all the lines I added are under 120. So is this failing because of previously written lines in the files I edited?

Could be that your local version of black needs updating, vscode isn't autoformatting correctly. You can use `python3.12 -m black --line-length 120 .` to run it locally. To update your version of black in the venv use `python3.12 -m pip install -e .[dev]` again.

File: test/e2e/test_catalogue_category.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rowan04) [reply test/e2e/test_catalogue_category.py]: fixed with added commit

**Review by joelvdavies — APPROVED** [review]
LGTM. I added a corresponding front end issue here https://github.com/ral-facilities/inventory-management-system/issues/1136.

**Pull request merged**